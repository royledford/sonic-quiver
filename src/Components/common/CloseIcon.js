import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class CloseIcon extends Component {
  static propTypes = {
    label: PropTypes.string.isRequired,
    size: PropTypes.number,
    lineWidth: PropTypes.number,
    color: PropTypes.string,
    className: PropTypes.string,
  }
  static defaultProps = {
    size: 16,
    lineWidth: 6,
    color: '#000',
    className: '',
  }

  render() {
    const { size, label, lineWidth, color, className } = this.props
    return (
      <div className={className}>
        <svg xmlns="http://www.w3.org/2000/svg" width={size} height={size} viewBox="0 0 39 39" aria-labelledby={label}>
          <title>{label}</title>
          <g>
            <line strokeWidth={lineWidth} stroke={color} x1="3" y1="36" x2="36" y2="3" />
            <line strokeWidth={lineWidth} stroke={color} x1="36" y1="36" x2="3" y2="3" />
          </g>
        </svg>
      </div>
    )
  }
}
