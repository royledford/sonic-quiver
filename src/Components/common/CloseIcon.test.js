import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

import CloseIcon from './CloseIcon'
import renderer from 'react-test-renderer'

describe('Close Icon', () => {
  it('renders', () => {
    const tree = renderer.create(<CloseIcon label="test" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('uses custom props', () => {
    const tree = renderer
      .create(<CloseIcon label="test label" size={16} lineWidth={6} color="#555" className="classname" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
