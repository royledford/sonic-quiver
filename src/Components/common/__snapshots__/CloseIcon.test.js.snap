// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Close Icon renders 1`] = `
<div
  className=""
>
  <svg
    aria-labelledby="test"
    height={16}
    viewBox="0 0 39 39"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      test
    </title>
    <g>
      <line
        stroke="#000"
        strokeWidth={6}
        x1="3"
        x2="36"
        y1="36"
        y2="3"
      />
      <line
        stroke="#000"
        strokeWidth={6}
        x1="36"
        x2="3"
        y1="36"
        y2="3"
      />
    </g>
  </svg>
</div>
`;

exports[`Close Icon uses custom props 1`] = `
<div
  className="classname"
>
  <svg
    aria-labelledby="test label"
    height={16}
    viewBox="0 0 39 39"
    width={16}
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      test label
    </title>
    <g>
      <line
        stroke="#555"
        strokeWidth={6}
        x1="3"
        x2="36"
        y1="36"
        y2="3"
      />
      <line
        stroke="#555"
        strokeWidth={6}
        x1="36"
        x2="3"
        y1="36"
        y2="3"
      />
    </g>
  </svg>
</div>
`;
